contracts/src/core/EigenDARegistryCoordinator.sol

```diff
@@ -150,91 +147,50 @@ contract EigenDARegistryCoordinator is
         }).numOperatorsPerQuorum;
 
         // For each quorum, validate that the new operator count does not exceed the maximum
-        // (If it does, an operator needs to be replaced -- see `registerOperatorWithChurn`)
-        for (uint256 i = 0; i < quorumNumbers.length; i++) {
+        // If it does, churns an operator via an exhaustive search through the operator set.
+        for (uint256 i; i < quorumNumbers.length; i++) {
             uint8 quorumNumber = uint8(quorumNumbers[i]);
 
-            require(
-                numOperatorsPerQuorum[i] <= _quorumParams[quorumNumber].maxOperatorCount,
-                "RegCoord.registerOperator: operator count exceeds maximum"
-            );
+            if (numOperatorsPerQuorum[i] > _quorumParams[quorumNumber].maxOperatorCount) {
+                _churnOperator(quorumNumber);
+            }
         }
     }
 
-    /**
-     * @notice Registers msg.sender as an operator for one or more quorums. If any quorum reaches its maximum operator
-     * capacity, `operatorKickParams` is used to replace an old operator with the new one.
-     * @param quorumNumbers is an ordered byte array containing the quorum numbers being registered for
-     * @param params contains the G1 & G2 public keys of the operator, and a signature proving their ownership
-     * @param operatorKickParams used to determine which operator is removed to maintain quorum capacity as the
-     * operator registers for quorums
-     * @param churnApproverSignature is the signature of the churnApprover over the `operatorKickParams`
-     * @param operatorSignature is the signature of the operator used by the AVS to register the operator in the delegation manager
-     * @dev `params` is ignored if the caller has previously registered a public key
-     * @dev `operatorSignature` is ignored if the operator's status is already REGISTERED
-     */
+    /// @notice Deprecated function. Use `registerOperator` instead.
```

Comment:
Might be good to add why its deprecated, explaining how we've moved from offchain churning to onchain. Also mention that we're keeping this function temporarily for backwards compatibility until operators have all upgraded to using the new offchain code, and then we'll be free to delete this function.